### PR TITLE
[profiler] Fix usage of the lock-free queue API.

### PR DESCRIPTION
**Pull request by alexrp**
* Free nodes hazardously.
* No need to unpoison nodes since we're not reusing them.
* Fix memory leak when emptying coverage queues.

### CODE REVIEW

File: mono/profiler/proflog.c [hunk 1]
```diff
@@ -3517,6 +3523,7 @@ count_queue (MonoLockFreeQueue *queue)
 	while ((node = mono_lock_free_queue_dequeue (queue))) {
 		count++;
 		mono_lock_free_queue_node_free (node);
+		mono_thread_hazardous_try_free (node, free_lfq_node);
```
Comment (kumpera) [inline mono/profiler/proflog.c]: mono_lock_free_queue_node_free must be called from the free func.

File: mono/profiler/proflog.c [hunk 2]
```diff
@@ -4352,7 +4359,7 @@ handle_writer_queue_entry (MonoProfiler *prof)
 
 		dump_buffer (prof, entry->buffer);
 
-		free (entry);
+		mono_thread_hazardous_try_free (entry, free_lfq_node);
```
Comment (kumpera) [inline mono/profiler/proflog.c]: Unless you operate on the entry hazardously, freeing it with HP makes little sense.

**Comment by kumpera** [comment]
LGTM

**Comment by alexrp** [comment]
@monojenkins merge

**Pull request merged**